ui/src/router/components/form/validation/schema.js

```diff
@@ -64,6 +64,14 @@ const routeSchema = yup.object().shape({
   timeout: timeoutSchema.required("Timeout is required"),
 });
 
+const validRouteSchema = yup
+  .mixed()
+  .test("valid-route", "Valid route is required", function (value) {
+    const configSchema = this.from.slice(-1).pop();
```

Comment:
The last item in the validation context has the entire router config (hierarchically), so using that to define this schema. This way, the schema can be used from multiple steps (fallback/final response validation and traffic rules).

Reply:
Thanks for refactoring this!